### PR TITLE
Solving erlings

### PR DESCRIPTION
**Pull request by metdinov**

### CODE REVIEW

File: sequential/bank_accounts/src/bank_account.erl [hunk 1]
```diff
@@ -2,5 +2,16 @@
 
 -export([process_operation/2]).
 
-process_operation(Bank, Operation) ->
-  put_your_solution_here.
+process_operation(Bank,
+		  {AccNumber, withdraw, Amount}) ->
```
Comment (facundoolano) [inline sequential/bank_accounts/src/bank_account.erl]: this indentation looks kind of strange

File: sequential/bank_accounts/src/bank_account.erl [hunk 2]
```diff
@@ -2,5 +2,16 @@
...
+    case lists:keyfind(AccNumber, 1, Bank) of
+      false -> {error, account_not_found};
```
Comment (facundoolano) [inline sequential/bank_accounts/src/bank_account.erl]: when doing case clauses, it's better for readability to put the implementations either all on the next line or all inline (if they're short enough), but not mix them

File: sequential/bank_accounts/src/bank_account.erl [hunk 3]
```diff
@@ -2,5 +2,16 @@
...
+      {AccNumber, Funds} when Funds < Amount ->
+	  {error, insufficient_funds};
+      {AccNumber, Funds} -> {AccNumber, Funds - Amount}
+    end;
+process_operation(Bank, {AccNumber, deposit, Amount}) ->
+    case lists:keyfind(AccNumber, 1, Bank) of
```
Comment (facundoolano) [inline sequential/bank_accounts/src/bank_account.erl]: you could do account finding (which is the same in both cases) in a helper function

File: sequential/bank_accounts/src/bank_account.erl (hunk 1 above)
Comment (metdinov) [reply sequential/bank_accounts/src/bank_account.erl]: My editor's formatter is killing me. I'll see to it.

File: sequential/bank_accounts/src/bank_account.erl (hunk 3 above)
Comment (metdinov) [reply sequential/bank_accounts/src/bank_account.erl]: Refactored 78aa5cce96a9d7c2bfdda7c93860d8119477a4ea

File: sequential/filter_fibonacci_numbers/src/filter_fibonacci_numbers.erl [hunk 4]
```diff
@@ -3,4 +3,13 @@
 -export([filter/1]).
 
 filter(List) ->
-  put_your_solution_here.
+    FibList = fib_less_than(lists:max(List)),
+    [X || X <- List, lists:member(X, FibList)].
+
+fib_less_than(N) ->
+    Fig_gen = fun Fib(I, J, Acc) ->
+		      if I =< N -> Fib(J, I + J, [I | Acc]);
```
Comment (facundoolano) [inline sequential/filter_fibonacci_numbers/src/filter_fibonacci_numbers.erl]: if is kind of weird in erlang, try to avoid it using case instead. BUT a top level case like this is better expressed with different function clauses and pattern match.

Also I, J, N are not great argument names, can you use semething more expressive like Current, Next and such>

Comment (facundoolano) [reply sequential/filter_fibonacci_numbers/src/filter_fibonacci_numbers.erl]: * More stuff: `Fig_gen` is not proper Erlang naming.
* I'd prefer you add another helper with the extra N argument instead of using a lambda fun here.

File: sequential/insert_element_at/src/insert_element_at.erl [hunk 5]
```diff
@@ -2,5 +2,15 @@
 
 -export([insert/3]).
 
-insert(_List, _Pos, _Element) ->
-  believe_in_yourself.
+insert(List, Pos, Element) ->
+    insert(List, Pos, Element, 1, []).
+
+insert([], _Pos, _Element, _Index, _Acc) -> [];
+insert([#{current := Current, old := _Old} | Tail], Pos,
```
Comment (facundoolano) [inline sequential/insert_element_at/src/insert_element_at.erl]: This code looks overly complex, unless I'm missing something you should be able to do this without a map and `++`. (a single list:reverse is probably ok)

Comment (facundoolano) [reply sequential/insert_element_at/src/insert_element_at.erl]: sorry I didn't see the description of the problem, disregard the part about the map

File: concurrent/priority/README.md [hunk 6]
```diff
@@ -8,7 +8,8 @@
 
 ## Exercise
 
-For this exercise we will process messages according to priorities from the erlang message box without using any data structure.
+For this exercise we will process messages according to priorities from the erlang message box without using any data structure.  
+Messages with priority `vip` should be processed _as soon as_ they enter the `priority_loop`. Carefully consider how messages are processed by a `receive` block, and how you can _combine_ more than one block to match clauses in a certain order. Hints on how to solve this found [here](http://learnyousomeerlang.com/more-on-multiprocessing).
```
Comment (AminArria) [inline concurrent/priority/README.md]: Make a separate PR for this. Remember we won't merge this one.

Comment (unbalancedparentheses) [reply concurrent/priority/README.md]: @metdinov please create a new PR with this.
